internal/controllers/machine/machine_controller.go

```diff
@@ -582,7 +582,11 @@ func (r *Reconciler) drainNode(ctx context.Context, cluster *clusterv1.Cluster,
 
 	restConfig, err := remote.RESTConfig(ctx, controllerName, r.Client, util.ObjectKey(cluster))
 	if err != nil {
-		log.Error(err, "Error creating a remote client while deleting Machine, won't retry")
+		if errors.Is(err, remote.ErrClusterLocked) {
```

Comment:
@killianmuldoon can we revert this part? remote.RESTConfig never returns this error

Reply:
You're right - I'll revert